xcat-inventory/xcclient/inventory/backend.py

```diff
@@ -0,0 +1,695 @@
+#!/usr/bin/python
+from __future__ import print_function
+import 	configparser
+import os
+import sh
+import shutil
+import re
+import sys
+from exceptions import * 
+import manager 
+import utils
+import pickle
+import hashlib
+
+class Invbackend:
+
+    globalcfgpath="/etc/xcat/inventory.cfg"
+    bkendcfg={}
+
+    def loadcfg(self,cfgpath=None):
+        if cfgpath is None:
+            myhome=utils.gethome()
```

Comment:
need to avoid it is `None' as we cannot ensure  `utils.gethome()` has a valid value.

Reply:
fixed